src/Core/Services/Implementations/OrganizationService.cs

```diff
@@ -2152,48 +2149,84 @@ private async Task ValidateOrganizationCustomPermissionsEnabledAsync(Guid organi
         }
     }
 
-    private async Task ValidateCustomPermissions(OrganizationUser user)
+    private async Task<bool> ValidateCustomPermissionsGrant(Guid organizationId, Permissions permissions)
```

Comment:
The old interface here was better - I think you should go back to passing in the [whole object](https://refactoring.guru/preserve-whole-object) (`OrganizationUser`). I assume you did it to fit the interface of `ValidateOrganizationUserUpdatePermissions`, but that too could be refactored to take the whole object, which would also be an improvement.

Reply:
That is what I intended to do but on `InviteUsersAsync` we do not have the `OrganizationUser` objects yet when calling `ValidateOrganizationUserUpdatePermissions` - they're only created afterwards.

Reply:
That's a good point. I think this is fine for now then.